qCC/ccCommandLineCommands.cpp

```diff
@@ -1457,18 +1457,17 @@ bool CommandSubsample::process(ccCommandLineInterface& cmd)
 				//calculate OCTREE level for each cloud based on required number of points
 				octreeLevel = CCCoreLib::DgmOctree::MAX_OCTREE_LEVEL;
 				unsigned numberOfPoints = sizeOfInputCloud;
-				//go through max->min untill previous ptsCount and current ptsCount is different then break;
-				do
+				//go through max->min until previous point count and current point count is different then break
+				for (int currentOctreeLevel = CCCoreLib::DgmOctree::MAX_OCTREE_LEVEL; currentOctreeLevel > 1; --currentOctreeLevel)
 				{
-					unsigned currentNumberOfPoints = octree->getCellNumber(octreeLevel);
-					octreeLevel--;
+					octreeLevel = currentOctreeLevel;
+					unsigned currentNumberOfPoints = octree->getCellNumber(currentOctreeLevel);
 					if (currentNumberOfPoints != numberOfPoints && numberOfPoints < maxNumberOfPoints)
 					{
 						break;
 					}
 					numberOfPoints = currentNumberOfPoints;
 				}
-				while (numberOfPoints > maxNumberOfPoints && octreeLevel > 0);
 				octreeLevel++;
```

Comment:
Why keeping this?

Reply:
you mean octreeLevel++? Because it will overiterate by one.

Reply:
In my edit I was testing that 'currentNumberOfPoints <= maxNumberOfPoints' (on line 1465).

So that it's faster and you don't need to add this additional increment.